app/views/common/SPKeyboardAvoidingView.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { KeyboardAvoidingView, Platform, StyleSheet } from 'react-native';
+
+const SPKeyboardAvoidingView = ({ behavior, children }) => {
+  const isiOS = Platform.OS === 'ios';
+
+  const instanceBehavior =
+    behavior === 'height' || behavior === 'position' || behavior === 'padding'
+      ? behavior
+      : null;
+
+  return (
+    <KeyboardAvoidingView
+      accessible={false}
+      style={styles.keyboardView}
+      autoScrollToFocusedInput
+      behavior={isiOS ? instanceBehavior : 'height'}>
+      {children}
+    </KeyboardAvoidingView>
+  );
+};
+
+const styles = StyleSheet.create({
+  keyboardView: {
+    flex: 1,
```

Comment:
We probably don't always want flex 1 here so I suggest moving that to the outer component and passing through style.

Though if you think we always need flex 1 then put a comment why here.. But also, allow style to be merged in

Reply:
since the logic here is that this component renders it's children. Leaving it to default flex will cause its children to shrink. Hence I kept it as 1 default. Without this 1 here the container will shrink and children won't display